### PR TITLE
formatting *.r code and better display of error messages about syntax problems for python and R

### PR DESCRIPTION
**Pull request by haraldschilly**
"formatR" with useful defaults and reporting python formatting errors (last line) if possible. also a bit of tweaking for the error message.

obligatory screenshots: messy R code, syntax error, formatted

screenshot from 2018-08-23 19-14-49

screenshot from 2018-08-23 20-44-15

screenshot from 2018-08-23 20-44-28

### CODE REVIEW

File: src/smc-project/browser-websocket/api.ts [hunk 1]
```diff
@@ -67,8 +72,13 @@ async function listing(path: string, hidden?: boolean): Promise<object[]> {
...
 import { run_prettier } from "../prettier";
-async function prettier(client: any, path: string, options: any): Promise<any> {
-  return await run_prettier(client, path, options);
+async function prettier(
+  client: any,
+  path: string,
+  options: any,
+  logger: any
+): Promise<any> {
+  return await run_prettier(client, path, options, logger);
```
Comment (williamstein) [inline src/smc-project/browser-websocket/api.ts]: This is a function that does NOTHING.  WEIRD.

File: src/smc-project/python-format.ts [hunk 2]
```diff
@@ -21,9 +21,21 @@ function yapf(input_path) {
   return spawn("yapf", ["-i", input_path]);
 }
 
+// TODO this doesn't do what it should …
```
Comment (williamstein) [inline src/smc-project/python-format.ts]: Please write a less mysterious comment.

File: src/smc-project/python-format.ts [hunk 3]
```diff
@@ -41,15 +53,18 @@ export async function python_format(
   py_formatter.stderr.on("data", data => (stderr += data.toString()));
   // wait for subprocess to close.
   let code = await callback(close, py_formatter);
+  // only last line
+  // stdout = last_line(stdout);
```
Comment (williamstein) [inline src/smc-project/python-format.ts]: why?  Not done?

**Comment by williamstein** [comment]
Code looks overall really good and safe to merge.  There's some weirdness, but it's mainly comments, which can wait for a followup PR.

File: src/smc-project/browser-websocket/api.ts (hunk 1 above)
Comment (haraldschilly) [reply src/smc-project/browser-websocket/api.ts]: yes, it was there like that before I added this. However, when I merged the one for jupyter cells, I removed it. see [merge commit](https://github.com/sagemathinc/cocalc/pull/3113/files#diff-05af434d8b71074e5c913dc6fad6d8dcR45)

**Pull request merged**